app/controllers/deliveries/new/select-recipient.js

```diff
@@ -0,0 +1,49 @@
+import Ember from 'ember';
+
+export default Ember.Controller.extend({
+  queryParams: ['projectId'],
+  projectId: null,
+  project: Ember.computed('projectId', function () {
+    return this.get('store').findRecord('duke-ds-project', this.get('projectId'));
+  }),
+  application: Ember.inject.controller(),
+  currentDukeDsUser: Ember.computed.alias('application.currentDukeDsUser'),
+  validUsersList: Ember.computed('model.[]', function () {
+    // remove users with invalid fullName values
+    return this.get('model')
+      .rejectBy('fullName', null)
+      .rejectBy('fullName', '(null)');
+  }),
+  otherUsersList: Ember.computed('validUsersList.[]', 'currentDukeDsUser', function () {
+    const currentDukeDSUser = this.get('currentDukeDsUser');
+    if (currentDukeDSUser) {
+      return this.get('validUsersList').rejectBy('id', currentDukeDSUser.get('id'));
+    } else {
+      return this.get('validUsersList');
+    }
+  }),
+  toUser: null,
+  disableNext: Ember.computed.not('toUser'),
+  actions: {
+    toUserSelectionChanged(actionData) {
+      var selectedItem = null;
+      if (actionData.selectedItems) {
```

Comment:
One benefit of Ember is that arrays (and other [enumerables](https://guides.emberjs.com/v2.18.0/object-model/enumerables/) have additional, observable properties like `firstObject` and `lastObject` that don't raise errors for empty arrays.

So, rather than having to check the length of the array first and then index into it, you could reduce the entire body to 

`this.set('toUser', selectedItems.get('firstObject'));`